### PR TITLE
RFC: cmake: Use `target_compile_features()`

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
Problem at hand: dependents (like FairRoot) need to know the minimum C++ standard level that FairMQ (and its headers) requires.

The first idea is to let the targets export their `CXX_STANDARD` value. First, this doesn't seem to work as expected.

Second, `target_compile_features()` seems to be the better way to go. It has a much better granularity, automatically has the export feature, and thus should make dependents behave correctly.

Also drop all of this enforeced CMAKE_CXX_STANDARD* setting. If it's given, check it. But that's it.

See: https://gitlab.kitware.com/cmake/cmake/-/issues/18446
See: https://cmake.org/cmake/help/latest/manual/cmake-compile-features.7.html#requiring-language-standards

* [X] Rebased against `dev` branch
* [X] My name is in the resp. CONTRIBUTORS/AUTHORS file
* [X] Followed [the seven rules of great commit messages](https://chris.beams.io/posts/git-commit/#seven-rules)

### CODE REVIEW

**Comment by ChristianTackeGSI** [comment]
This is not necassarily the final or good version of this.

This is a version, that works for me.

And I want to start some discussion on this.

File: cmake/FairMQLib.cmake
Comment (dennisklein) [inline cmake/FairMQLib.cmake:98]:
```suggestion
  if(CMAKE_CXX_STANDARD AND CMAKE_CXX_STANDARD VERSION_LESS PROJECT_MIN_CXX_STANDARD)
```

**Review by dennisklein — COMMENTED** [review]
Looks fine. Need to check how this behaves with FairRoot and AliceO2 (in general projects that set `CMAKE_CXX_STANDARD` in addition) before merging.

**Comment by ChristianTackeGSI** [comment]
> Need to check how this behaves with FairRoot

FairRoot has a bit questionable setup, because it does not set `*CXX_STANDARD ` at all.:
```console
FairRoot$ grep CXX_STANDARD -r .
FairRoot$
```

In contrast, it has code that forces people to pass in `-std=` flags by hand, so that cmake has no chance of analyzing them correctly:
https://github.com/FairRootGroup/FairRoot/blob/dev/CMakeLists.txt#L115-L124

This yields to the manually passed flag and the flag created by cmake to both end up in the final compiler call.

I will look into this deeper tomorrow.

> and AliceO2 (in general projects that set CMAKE_CXX_STANDARD in addition)

Who could we ping to check this for AliceO2?

My general hope is, that cmake just does the right thing when a target has `CXX_STANDARD` specified (via CMAKE_CXX_STANDARD) and in addition an imported target also has compiler_features set. I would assume, that it just switches the compiler to a mode that will satisfy everything.

**Review by dennisklein — APPROVED** [review]
will merge once I fixed the CI

**Comment by dennisklein** [comment]
cherry-picked in #363

**Pull request closed without merging**
